5.2_join.py

```diff
@@ -1,13 +1,17 @@
 def join(*lists, sep='-'):
-    # function that receives unlimited lists and if gets also seperator parameter
-    # combines them with seperator between each list.
-    # the function get Unlimited lists and Seperator - separate between each list
-    # function returns the list with seperator in the end (without the last character)
-    new_list = [lst for curr_list in lists for lst in curr_list + [sep]][:-1]
-    if new_list == []:
+    """
+    function that receives unlimited lists and if gets also seperator parameter
+    combines them with seperator between each list.
+    :param lists: unlimited lists
+    :param sep: seperator parameter (optional, default- ('-').
+    :return: list with seperator in the end (without the last character)
```

Comment:
Why without the last character?
sep should be concatenated between every two list, it is not supposed to replace the last character in every list.